R/pkg/R/context.R

```diff
@@ -225,6 +225,37 @@ setCheckpointDir <- function(sc, dirName) {
   invisible(callJMethod(sc, "setCheckpointDir", suppressWarnings(normalizePath(dirName))))
 }
 
+#' Add a file or directory to be downloaded with this Spark job on every node.
+#'
+#' The path passed can be either a local file, a file in HDFS (or other Hadoop-supported
+#' filesystems), or an HTTP, HTTPS or FTP URI.  To access the file in Spark jobs,
+#' use sparkFiles.get(fileName) to find its download location.
+#'
+#' A directory can be given if the recursive option is set to true.
+#' Currently directories are only supported for Hadoop-supported filesystems.
+#'
+#' @param path The path of the files to be added
+#' @param recursive Recursive or not if the path is directory. Default is FALSE.
+#' @noRd
+#' @examples
+#'\dontrun{
+#' sc <- sparkR.init()
+#' addFile(sc, "myfile")
+#'}
+addFile <- function(sc, path) {
+  invisible(callJMethod(sc, "addFile", suppressWarnings(normalizePath(path))))
```

Comment:
Otherwise, I think we can not just `normalizePath` here to avoid this problem. I will let you know after testing if that would really avoid this problem.